lib/rubocop/cop/style/redundant_begin.rb

```diff
@@ -46,9 +46,11 @@ def autocorrect(node)
 
             corrector.replace(
               range_with_surrounding_space(node.loc.expression),
-              range_with_surrounding_space(
-                child.loc.expression
-              ).source.gsub(/^[ \t]{#{indent_diff}}/, '')
+              range_with_surrounding_space(node.loc.expression)
+                .source
+                .gsub(/\A\s*?begin\s*?(;|$)/, '')
```

Comment:
Agreed. My first two failed attempts to eliminate this destructive behavior manipulated the nodes directly. Maybe someone with more knowledge of the parsing library, where I believe the root of the problem lies, could suggest an alternative solution?